Use bitmap to represent SFP errors

Junchao-Mellanox opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->

#### Description
<!--
     Describe your changes in detail
-->

#### Motivation and Context
<!--
     Why is this change required? What problem does it solve?
     If this pull request closes/resolves an open Issue, make sure you
     include the text "fixes #xxxx", "closes #xxxx" or "resolves #xxxx" here
-->

#### How Has This Been Tested?
<!--
     Please describe in detail how you tested your changes.
     Include details of your testing environment, and the tests you ran to
     see how your change affects other areas of the code, etc.
-->

#### Additional Information (Optional)

Junchao-Mellanox committed:
Add bitmap support for SFP error event

Junchao-Mellanox committed:
Add unit test for sfp error bitmap

keboliu commented on sonic-xcvrd/xcvrd/xcvrd.py:
```diff
@@ -47,13 +47,14 @@
+    8: 'SFP_STATUS_ERR_UNSUPPORTED_CABLE',
+    16: 'SFP_STATUS_ERR_HIGH_TEMP',
+    32: 'SFP_STATUS_ERR_BAD_CABLE'
+}
```
maybe we still need to keep a set of "errors_block_eeprom_reading" as a subset of the whole errors, meaning not all errors will block the EEPROM reading, only on some certain error we will stop reading the EEPROM, but on others we will still continue to read.

Junchao-Mellanox replied:
According to previous code implementation, all kinds of errors will be treated as "blocking eeprom read error". I suppose there will be no new errors in short term, could we keep it this way for now?

Junchao-Mellanox replied:
@keboliu please let me know if you still have concern about this.

Junchao-Mellanox committed:
Move common code to sfp_status_helper.py

keboliu commented on sonic-xcvrd/tests/test_xcvrd.py:
```diff
@@ -24,13 +24,15 @@
 modules_path = os.path.dirname(test_path)
 scripts_path = os.path.join(modules_path, "xcvrd")
-helper_file_path = os.path.join(scripts_path, "xcvrd_utilities"+"/y_cable_helper.py")
+#helper_file_path = os.path.join(scripts_path, "xcvrd_utilities"+"/y_cable_helper.py")
```
remove this line if not needed.

keboliu commented on sonic-xcvrd/tests/test_xcvrd.py:
```diff
@@ -24,13 +24,15 @@
 os.environ["XCVRD_UNIT_TESTING"] = "1"
-load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
-from y_cable_helper import *
+#load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
```
remove below two lines if not needed

Junchao-Mellanox committed:
Remove unused comment lines

Commit:
Fix LGTM warning

Junchao-Mellanox committed:
Fix LGTM warning

Junchao-Mellanox committed:
Merge branch 'sfp-bit-map' of github.com:Junchao-Mellanox/sonic-platform-daemons into sfp-bit-map

Junchao-Mellanox committed:
Remove unused import

stephenxs committed:
Handle the error status returned by platform APIs

In case there is an error returned by platform API
- Translate the error bitmap to error description for generic errors
- Fetch the error description from sfp_error dict or via calling platform API get_error_description for vendor specific errors
- Adjust unit test cases

Signed-off-by: Stephen Sun <stephens@nvidia.com>

stephenxs committed:
Fix review comments

Signed-off-by: Stephen Sun <stephens@nvidia.com>

Junchao-Mellanox committed:
Merge pull request #26 from stephenxs/sfp-bit-map-error-status

Enhanced - Handle the error status returned by platform APIs
